wagtail/models/__init__.py

```diff
@@ -734,20 +755,19 @@ def _get_dummy_headers(self, original_request=None):
         if url:
             url_info = urlsplit(url)
             hostname = url_info.hostname
+            if not validate_host(
+                hostname,
+                settings.ALLOWED_HOSTS or [".localhost", "127.0.0.1", "[::1]"],
+            ):
+                # The hostname is not valid according to ALLOWED_HOSTS - use a fallback
+                hostname = self._get_fallback_hostname()
```

Comment:
I was slightly wary about this change, but then I remembered the proper validation for the actual request should already happen at the real view level. This method is only used to simulate the page's view mechanism so we can get the rendered output to be the real view's response. So I think we should be OK.